internal/provider/dedicated_cluster_resource.go

```diff
@@ -0,0 +1,792 @@
+			}
+		}
+	}
+
+	data.TiKVNodeSetting = tikvNodeSetting{
+		NodeSpecKey:         types.StringValue(resp.TikvNodeSetting.NodeSpecKey),
+		NodeCount:           types.Int32Value(resp.TikvNodeSetting.NodeCount),
+		StorageSizeGi:       types.Int32Value(resp.TikvNodeSetting.StorageSizeGi),
+		StorageType:         types.StringValue(string(*resp.TikvNodeSetting.StorageType)),
+		NodeSpecDisplayName: types.StringValue(*resp.TikvNodeSetting.NodeSpecDisplayName),
+	}
+
+	// may return
+	// tiflash node setting
+	if resp.TiflashNodeSetting != nil {
+		data.TiFlashNodeSetting = &tiflashNodeSetting{
+			NodeSpecKey:         types.StringValue(resp.TiflashNodeSetting.NodeSpecKey),
+			NodeCount:           types.Int32Value(resp.TiflashNodeSetting.NodeCount),
+			StorageSizeGi:       types.Int32Value(resp.TiflashNodeSetting.StorageSizeGi),
+			StorageType:         types.StringValue(string(*resp.TiflashNodeSetting.StorageType)),
+			NodeSpecDisplayName: types.StringValue(*resp.TiflashNodeSetting.NodeSpecDisplayName),
+		}
+	}
+}
+
+func (r dedicatedClusterResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// get plan
+	var plan dedicatedClusterResourceData
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// get state
+	var state dedicatedClusterResourceData
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Check if paused state is changing
+	isPauseStateChanging := plan.Paused.ValueBool() != state.Paused.ValueBool()
+
+	// Check if TiFlashNodeSetting is changing
+	isTiFlashNodeSettingChanging := false
+	if plan.TiFlashNodeSetting != nil && state.TiFlashNodeSetting != nil {
+		isTiFlashNodeSettingChanging = plan.TiFlashNodeSetting.NodeCount != state.TiFlashNodeSetting.NodeCount ||
+			plan.TiFlashNodeSetting.NodeSpecKey != state.TiFlashNodeSetting.NodeSpecKey ||
+			plan.TiFlashNodeSetting.StorageSizeGi != state.TiFlashNodeSetting.StorageSizeGi
+	} else if plan.TiFlashNodeSetting != nil {
+		isTiFlashNodeSettingChanging = true
+	} else if state.TiFlashNodeSetting != nil {
+		isTiFlashNodeSettingChanging = true
+	}
+
+	// Check if any other attributes are changing
+	isOtherAttributesChanging := (plan.DisplayName != state.DisplayName ||
+		plan.TiDBNodeSetting.NodeCount != state.TiDBNodeSetting.NodeCount ||
+		plan.TiDBNodeSetting.NodeSpecKey != state.TiDBNodeSetting.NodeSpecKey ||
+
+		plan.TiKVNodeSetting.NodeCount != state.TiKVNodeSetting.NodeCount ||
+		plan.TiKVNodeSetting.NodeSpecKey != state.TiKVNodeSetting.NodeSpecKey ||
+		plan.TiKVNodeSetting.StorageSizeGi != state.TiKVNodeSetting.StorageSizeGi ||
+
+		!plan.Labels.Equal(state.Labels)) ||
+		plan.RootPassword != state.RootPassword ||
+		isTiFlashNodeSettingChanging
+
+	// If trying to change pause state along with other attributes, return an error
+	if isPauseStateChanging && isOtherAttributesChanging {
```

Comment:
how about istiflashchanging?

Reply:
I don't get you, `isTiFlashNodeSettingChanging` is included in `isOtherAttributesChanging`. Do you attempt to rename the variable?

Reply:
ahh, I missed it.